src/lib/plugin-loader.coffee

```diff
@@ -15,50 +15,104 @@ safefs = require('safefs')
 # Classes
 
 # Define Plugin Loader
+###*
+# The Plugin Loader class
+# @class PluginLoader
+# @constructor
+###
 class PluginLoader
 
 	# ---------------------------------
 	# Constructed
 
-	# DocPad Instance
+	###*
+	# The DocPad Instance
+	# @private
+	# @property {Object} docpad
+	###
 	docpad: null
 
-	# BasePlugin Class
+
+	###*
+	# The BasePlugin Class
+	# @private
+	# @property {Object}
+	###
 	BasePlugin: null
 
+
+	###*
 	# The full path of the plugin's directory
+	# @private
+	# @property {String}
+	###
 	dirPath: null
 
 
 	# ---------------------------------
 	# Loaded
 
+	###*
 	# The full path of the plugin's package.json file
+	# @private
+	# @property {String}
+	###
 	packagePath: null
 
+	###*
 	# The parsed contents of the plugin's package.json file
+	# @private
+	# @property {Object}
+	###
 	packageData: {}
 
+	###*
 	# The full path of the plugin's main file
+	# @private
+	# @property {String}
+	###
 	pluginPath: null
 
+
+	###*
 	# The parsed content of the plugin's main file
+	# @private
+	# @property {Object}
+	###
 	pluginClass: {}
 
-	# Plugin name
+	###*
+	# The plugin name
+	# @private
+	# @property {String}
+	###
 	pluginName: null
 
-	# Plugin version
+	###*
+	# The plugin version
+	# @private
+	# @property {String}
+	###
 	pluginVersion: null
 
+	###*
 	# Node modules path
+	# @private
+	# @property {String}
+	###
 	nodeModulesPath: null
 
 
 	# ---------------------------------
 	# Functions
 
-	# Constructor
+	###*
+	# Constructor method
+	# @method constructor
+	# @param {Object} @docpad
+	# @param {String} @dirPath
+	# @param {Object} @BasePlugin
+	###
 	constructor: ({@docpad,@dirPath,@BasePlugin}) ->
```

Comment:
Is that the correct convention, as what this code does is say we receive a opts argument, that contains `docpad`, `dirPath`, and `basePlugin` properties, which are then assigned to `this`